src/db.rs

```diff
@@ -405,9 +419,20 @@ impl DbInner {
     }
 
     async fn flush_wals(&self) -> Result<(), SlateDBError> {
+        self.send_flush_wals(true).await
+    }
+
+    async fn await_flush_wals(&self) -> Result<(), SlateDBError> {
+        self.send_flush_wals(false).await
```

Comment:
I think we can make the waiting mechanism here much simpler, and just do what the writes do when run with `await_durable: true`:

```
        let wal = {
            let guard = self.state.read();
            let snapshot = guard.snapshot();
            snapshot.wal.clone()
        };
        if wal.is_empty() {
            return Ok(())
        }
        wal.await_durable().await
```

That should also simplify the changes to the flush/memtable tasks since they won't have to track a bunch of waiters.

Reply:
Ah nice. Yeah, that makes things simpler.